### PR TITLE
Index ```Search Template``` API tests

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
test for:
GET /{index}/_search/template
POST /{index}/_search/template
GET /_search/template
POST /_search/template

### Issues Resolved
[#663]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/_core.yaml [hunk 1]
```diff
@@ -3266,6 +3266,8 @@ components:
                     $ref: '../schemas/_core.search.yaml#/components/schemas/Suggest'
               terminated_early:
                 type: boolean
+              status:
```
Comment (dblock) [inline spec/namespaces/_core.yaml:3269]: This should go into CHANGELOG in fixes as "Added missing `status` to `/_search/template` response"

**Review by dblock — CHANGES_REQUESTED** [review]
Looks good, do add to CHANGELOG.

**Comment by Tokesh** [comment]
@dblock ready for your review!

**Comment by Tokesh** [comment]
> Sorry to be annoying. Let's put these tests in the right place?
> 
> `/{index}/_search/template` -> `tests/default/indices/search/template.yaml` `/_search/template` -> `tests/default/search/template.yaml`

Its doesn't annoying! Thank you

**Review by dblock — APPROVED** [review]

**Pull request merged**